### PR TITLE
jetpack-mu-wpcom: Improve Verbum build

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

* Avoid copying php files into `src/build/`, as it causes Composer to complain about multiple files providing the classes.
* Run `dynamic-loader.js` through webpack for minification.
* Use the jetpack-assets package to register the scripts using the data in the `.assets.php` files.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1706559158287209/1706556445.671539-slack-C01U2KGS2PQ

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
1. Apply this patch to your sandbox using both of the commands listed [here](https://github.com/Automattic/jetpack/pull/35323#issuecomment-1915666258).
2. Sandbox punsintended418746564.wordpress.com.
3. Comment on this [post](https://punsintended418746564.wordpress.com/2023/12/12/fresh-post).
4. Make sure embeds works (one embed is enough, to make sure embeds endpoint work (relies on PHP files)).

### Regression testing
1. Checkout this branch.
2. Sync jetpack-mu-plugin with your atomic site.
3. Visit any wp-admin page and verify no warnings or errors show up.

### CODE REVIEW

**Comment by anomiex** [comment]
Hi Verbum folks! Sorry for dumping a PR on you without doing much testing myself. :slightly_frowning_face: I don't really know how Verbum works. Please feel free to fix stuff if there are obvious errors in the PR, and if this works for you feel free to merge it. Thanks!

**Review by alshakero — APPROVED** [review]
Thank you so much! 

1. I added testing steps for future ref.
2. I verified `dynamic-loader.js` is versioned and minified.
3. I verified Verbum works, including the block editor.
4. I verified the embeds work. 
5. I tested on Atomic and I don't see any warnings or errors.

File: projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php [hunk 1]
```diff
@@ -250,14 +246,14 @@ public function enqueue_assets() {
 
 		wp_enqueue_script( 'verbum-settings' );
 
-		wp_enqueue_script(
+		Assets::register_script(
```
Comment (alshakero) [inline projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php:249]: cc @Automattic/vertex it seems that's why we had issues with our versioning! Let's fix this [one](https://github.com/Automattic/jetpack/blob/add/sso-improvements/projects/plugins/jetpack/modules/sso.php#L343) in our SSO work.

**Review by agrullon95 — APPROVED** [review]
## Testing:

Synced changes to WoA site and did not see any warnings/errors.
Verified embed blocks are still working.
Tested posts work for WPCOM, FB, and anon users with and without GB editor (Simple and Atomic).
Verified dynamic-loader.js is versioned and minified.

File: projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php (hunk 1 above)
Comment (escapemanuele) [reply projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php:249]: On it 👍

**Pull request merged**